### ISSUE TITLE
Question about equation (18) in the paper

### ISSUE THREAD

**Issue by leleogere** [YihongDong/CODEP#3]
Hello,

I took some time to read your paper, thank you for this work!

However, I have some trouble understanding the equation (18):
![Equation (18)](https://github.com/user-attachments/assets/d9c8c26d-88fb-459a-84f9-4a596c02af2f)

In this equation, $\hat{s_t}$ represents the actual state of the PDA module. However, I do not understand how you get the value of $p(\hat{s_t} \mid y_{<t}, s_{<t}, x)$.

More generally, I have trouble understanding how you derive the complete equation (18) from the equations (15), (16) and (17).

I would be grateful if you could enlighten me on those points!

Thank you.

**Comment by YihongDong** [comment]
Thank you for your attention to our work. Equation (18) represents the probabilities computed during the inference phase, while Equations (15), (16), and (17) correspond to the loss functions computed during the training phase. We define p(y'_t) as the Joint Prediction with State in Equation (18), which is a weighted sum of p(y_t) and p(\hat{s}_t).

**Comment by leleogere** [comment]
Thank you for your answer. I get that (18) is a weighted sum of $p(y_t)$ and $p(\hat{s_t})$, but my 2 questions are the following:
- How do you derive this weighted sum? How are the coefficients $\frac{1}{1+\alpha}$ and $\frac{\alpha}{1+\alpha}$ chosen?
- How do you get the value of $p(\hat{s_t})$? The model computes the probability distribution $p(s_t)$, not $p(\hat{s_t})$, does it?

**Comment by YihongDong** [comment]
For the first question, the results of the weighted sum should lie within the interval $(0, 1)$, and the coefficients are selected intuitively. For the second question, $\hat{s_t}$ represents the actual state of the PDA module, which is determined by the function $\delta(s_{t-1}, g_{t-1}, y_t)$, we compute the probability distribution $p$ and obtain the value of $p(\hat{s_t})$.

**Comment by leleogere** [comment]
Thank you for the first question!

For the second question, if I understand correctly, for each (valid) token $$y_t$$, you compute the state that would the PDA be in with this token produced (that is called $$\hat{s_t}$$), you take the probability for this specific state in the distribution predicted by your model, and you compute the final probability of that studied token $$y'_t$$ using the weighted sum right? That way, tokens with low probas but leading into a probable state get assigned a higher probability, and tokens with a high proba but leading into an unlikely state get assigned a lower probability, correct?

**Comment by YihongDong** [comment]
Yes, your understanding is correct!

**Comment by leleogere** [comment]
Thank you for your explanation!
